sdk/core/core-amqp/src/retry.ts

```diff
@@ -277,6 +277,19 @@ export async function retry<T>(config: RetryConfig<T>): Promise<T> {
   if (success) {
     return result;
   } else {
-    throw lastError;
+    throw compileErrors(errors);
   }
 }
+
+function compileErrors(errors: (MessagingError | Error)[]): MessagingError | Error {
+  if (!errors.length) {
+    throw new Error("Impossible to happen");
```

Comment:
I can see why, but perhaps we could throw a more descriptive error just in case?

```suggestion
    throw new RangeError("Error array is empty");
```